bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -248,23 +314,69 @@ protected NeoHubReadDcbResponse fromNeoHubReadDcbResponse() {
      * handlers
      */
     private synchronized void lazyPollingSchedulerExecute() {
-        NeoHubInfoResponse infoResponse = fromNeoHubReadInfoResponse();
+        // check which API is supported
+        if (!isApiOnline) {
+            selectApi();
+        }
 
-        if (infoResponse != null) {
-            // determine temperatureUnit
-            NeoHubReadDcbResponse dcbResponse = fromNeoHubReadDcbResponse();
-            Unit<?> temperatureUnit = (dcbResponse != null) ? dcbResponse.getTemperatureUnit() : SIUnits.CELSIUS;
+        NeoHubAbstractDeviceData deviceData = fromNeoHubGetDeviceData();
+        if (deviceData != null) {
+            // check the temperature unit
+            if (systemDataDirty) {
+                NeoHubReadDcbResponse systemData = fromNeoHubReadSystemData();
+                if (systemData != null) {
+                    temperatureUnit = systemData.getTemperatureUnit();
+                    systemDataDirty = false;
+                    logger.info("Setting temperature unit: => {}", temperatureUnit);
+                }
+            }
 
-            // dispatch infoResponse to each of the hub's owned devices ..
+            // dispatch deviceData to each of the hub's owned devices ..
             List<Thing> children = getThing().getThings();
             for (Thing child : children) {
                 ThingHandler device = child.getHandler();
                 if (device instanceof NeoBaseHandler) {
-                    ((NeoBaseHandler) device).toBaseSendPollResponse(infoResponse, temperatureUnit);
+                    ((NeoBaseHandler) device).toBaseSendPollResponse(deviceData);
                 }
             }
-        }
 
+            // evaluate and update the state of our RF mesh QoS channel
+            List<? extends AbstractRecord> devices = deviceData.getDevices();
+            State state;
+
+            if (devices == null || devices.isEmpty()) {
+                state = UnDefType.UNDEF;
+            } else {
+                int totalDeviceCount;
+
+                if ((totalDeviceCount = devices.size()) == 0) {
+                    state = UnDefType.UNDEF;
+                } else {
```

Comment:
I am not sure I understand you. On one hand I am checking if devices is null and on the other hand I am checking if it is empty..

Reply:
The QoS measurement is a percentage of devices that are actually online in the RF mesh, divided by devices that have been configured in the hub. Obviously if there are zero devices configured in the hub, this would result in a divide-by-zero exception, so in that special case, we return UNDEF instead.

Reply:
Ah. Ok. I see my error now. Sorry for the confusion.